packages/flutter_driver/lib/src/extension.dart

```diff
@@ -271,4 +274,27 @@ class FlutterDriverExtension {
     _frameSync = setFrameSyncCommand.enabled;
     return new SetFrameSyncResult();
   }
+
+  SemanticsHandle _semantics;
+
+  Future<SetSemanticsResult> _setSemantics(Command command) async {
+    final SetSemantics setSemanticsCommand = command;
+    final bool wasEnabled = RendererBinding.instance.pipelineOwner.semanticsOwner != null;
+    if (setSemanticsCommand.enabled && _semantics == null) {
+      _semantics = RendererBinding.instance.pipelineOwner.ensureSemantics();
+      if (!wasEnabled) {
+        // wait for the first frame where semantics is enabled.
+        final Completer<Null> completer = new Completer<Null>();
+        SchedulerBinding.instance.addPostFrameCallback((Duration d) {
+          completer.complete();
+        });
+        await completer.future;
+      }
+    } else if (!setSemanticsCommand.enabled && _semantics != null) {
+      _semantics.dispose();
+      _semantics = null;
+    }
+    final bool enabled = RendererBinding.instance.pipelineOwner.semanticsOwner != null;
```

Comment:
nit: since the other variable is called `wasEnabled`, perhaps this one should be called `isEnabled`. But also, you could make the expression into a getter, then you wouldn't need a second variable:

```dart
bool get isEnabled => RendererBinding.instance.pipelineOwner.semanticsOwner != null;
bool wasEnabled = isEnabled;
...
return new SetSemanticsResult(wasEnabled != isEnabled);
```